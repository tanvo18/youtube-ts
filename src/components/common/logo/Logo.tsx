import React from 'react'
import './logo.scss';

interface LogoProps {
  logoClass: string;
  logoImg: string;
  link: string;
  logoAlt: string;
}

const Logo: React.FC<LogoProps> = ({ logoClass, logoImg, link, logoAlt }) => {
  return (
    <div>
      <h1
        className={logoClass}
      >
        <a href={link}>
          <img src={logoImg} alt={logoAlt} />
        </a>
      </h1>
    </div>
  )
}

export default Logo
