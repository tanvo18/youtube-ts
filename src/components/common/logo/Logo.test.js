import React from 'react';
import Logo from './Logo';
import renderer from 'react-test-renderer';

const props = {
  logoClass: 'logo',
  logoImg: '../../assets/images/youtube.png',
  logoAlt: 'this is a logo',
  link: '/'
};

describe('test logo', () => {
  it('should render logo source correctly', () => {
    const wrapper = shallow(<Logo {...props} />);
    expect(wrapper.find('img').props().src).toBe('../../assets/images/youtube.png');
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<Logo {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
