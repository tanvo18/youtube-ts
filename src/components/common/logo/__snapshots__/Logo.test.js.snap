// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test logo should render correctly 1`] = `
<div>
  <h1
    className="logo"
  >
    <a
      href="/"
    >
      <img
        alt="this is a logo"
        src="../../assets/images/youtube.png"
      />
    </a>
  </h1>
</div>
`;
